### PR TITLE
remove cloneDeep call

### PR DESCRIPTION
**Pull request by nmalyschkin**

### CODE REVIEW

File: packages/esm-extensions/src/render.ts [hunk 1]
```diff
@@ -16,6 +14,18 @@ export interface CancelLoading {
   (): void;
 }
 
+const immutableSet = <T extends Record<string, any>>(
+  obj: T,
+  [parm, ...resPath]: string[],
```
Comment (FlorianRappl) [inline packages/esm-extensions/src/render.ts:19]: Can we use `Array<string>`? What is `parm` and `resPath` (are better / more descriptive names viable here)?

**Comment by brandones** [comment]
This antipattern exists a handful of times in the codebase (I think). Maybe we can add this as a helper function to `esm-state`?

And can we call it just `set`? Or `update` following the example of [React's update](https://reactjs.org/docs/update.html)?

File: packages/esm-extensions/src/render.ts [hunk 2]
```diff
@@ -16,6 +14,18 @@ export interface CancelLoading {
   (): void;
 }
 
+const immutableSet = <T extends Record<string, any>>(
```
Comment (FlorianRappl) [inline packages/esm-extensions/src/render.ts:17]: I guess this could / should be a `function`

**Comment by FlorianRappl** [comment]
> 
> 
> This antipattern exists a handful of times in the codebase (I think). Maybe we can add this as a helper function to `esm-state`?
> 
> And can we call it just `set`? Or `update` following the example of [React's update](https://reactjs.org/docs/update.html)? Or (if needed) `immutablySet`? `immutableSet` sounds like an object.

Sounds good to me. `update` and placed in `esm-state` has my vote.

**Review by FlorianRappl — APPROVED** [review]

**Pull request merged**